i18n-helpers/src/directives.rs

```diff
@@ -0,0 +1,104 @@
+use regex::Regex;
+use std::sync::OnceLock;
+
+#[derive(Debug, PartialEq)]
+pub enum Directive {
+    Skip,
+    TranslatorComment(String),
+}
+
+pub fn find(html: &str) -> Option<Directive> {
+    static RE: OnceLock<Regex> = OnceLock::new();
+    let re = RE.get_or_init(|| {
+        let pattern = r"(?x)
+              <!-{2,}\s*                  # the opening of the comment
+              (?:i18n|mdbook-xgettext)    # the prefix
+              \s*:                        # delimit between prefix and command
+              (?<command>.*[^-])          # the command part of the prefix
+              -{2,}>                      # the closing of the comment
+        ";
+        Regex::new(pattern).expect("well-formed regex")
+    });
+
+    let captures = re.captures(html.trim())?;
+
+    let command = captures["command"].trim();
+    match command.split(is_delimiter).next() {
+        Some("skip") => Some(Directive::Skip),
+        Some("comment") => {
+            let start_of_comment_offset = std::cmp::min(
```

Comment:
Oh, I finally understand why we have this complexity: we're accepting things like

```
<!-- i18n: skip -->
<!-- i18n: comment foo bar -->
```

where you parse the first word of the `comment foo bar` part if you see that this is a comment.

If I got that right, then I suggest we change our minds :grimacing: and make this less ambigious. Something like this:

```
<!-- i18n-skip -->
<!-- i18n: foo bar -->
```

I'm thinking of a `function-name: arguments` syntax here. So `i18n-skip` would be a recognized "function" and it takes no arguments. Similarly, `i18n` would be a recognized function and its arguments is a translator comment.

We can merge this already now and then change our minds as we get a bit of experience with this.

Reply:
Agreed; let 's merge first, and then I will revise to use `function-name: arguments`.